src/injected/analyzers/tab-stops-requirement-result-processor.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { BaseStore } from 'common/base-store';
+import { FeatureFlags } from 'common/feature-flags';
+import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
+import { VisualizationScanResultData } from 'common/types/store-data/visualization-scan-result-data';
+import { TabStopRequirementActionMessageCreator } from 'DetailsView/actions/tab-stop-requirement-action-message-creator';
+import { AllFrameRunner } from 'injected/all-frame-runner';
+import { AutomatedTabStopRequirementResult } from 'injected/tab-stop-requirement-result';
+import { isEqual } from 'lodash';
+
+export class TabStopsRequirementResultProcessor {
+    private seenTabStopRequirementResults: AutomatedTabStopRequirementResult[] = [];
+    private isStopped: boolean = true;
+    private needsRequirementRunner: boolean;
+
+    constructor(
+        private readonly featureFlagStore: BaseStore<FeatureFlagStoreData>,
+        private readonly tabStopRequirementRunner: AllFrameRunner<AutomatedTabStopRequirementResult>,
+        private readonly tabStopRequirementActionMessageCreator: TabStopRequirementActionMessageCreator,
+        private readonly visualizationResultsStore: BaseStore<VisualizationScanResultData>,
+    ) {}
+
+    public start = (needsRequirementRunner: boolean): void => {
+        if (!this.isStopped) {
+            return;
+        }
+
+        this.needsRequirementRunner = needsRequirementRunner;
+
+        if (
+            this.needsRequirementRunner &&
+            this.featureFlagStore.getState()[FeatureFlags.tabStopsAutomation] === true
+        ) {
+            this.seenTabStopRequirementResults = [];
+            this.tabStopRequirementRunner.topWindowCallback = this.processTabStopRequirementResults;
+            this.tabStopRequirementRunner.start();
+        }
+
+        this.isStopped = false;
+    };
+
+    public listenToStore(): void {
+        this.visualizationResultsStore.addChangedListener(this.onStateChange);
+        this.featureFlagStore.addChangedListener(this.onStateChange);
+        this.onStateChange();
+    }
+
+    private onStateChange = (): void => {
+        const state = this.visualizationResultsStore.getState();
+        if (state.tabStops.tabbingCompleted && state.tabStops.needToCollectTabbingResults) {
+            this.stop();
+        }
+    };
+
+    public stop = (): void => {
+        if (this.isStopped) {
+            return;
+        }
+
+        if (this.needsRequirementRunner) {
+            this.tabStopRequirementRunner.stop();
+        }
+
+        const state = this.visualizationResultsStore.getState();
+        if (state.tabStops.tabbingCompleted && state.tabStops.needToCollectTabbingResults) {
```

Comment:
What's the scenario where this condition is not true? Or specifically, a scenario where we would be calling stop but would not want us to make these calls to the action message creator.

Reply:
We want to send telemetry only after the first time the user finished tabbing and then stop collecting tabbing results